src/dbnode/storage/index.go

```diff
@@ -935,11 +946,19 @@ func (i *nsIndex) flushableBlocks(
 func (i *nsIndex) canFlushBlock(
 	block index.Block,
 	shards []databaseShard,
+	flushType series.WriteType,
 ) (bool, error) {
 	// Check the block needs flushing because it is sealed and has
-	// any mutable segments that need to be evicted from memory
-	if !block.IsSealed() || !block.NeedsMutableSegmentsEvicted() {
-		return false, nil
+	// any mutable/cold mutable segments that need to be evicted from memory.
+	switch flushType {
+	case series.WarmWrite:
+		if !block.IsSealed() || !block.NeedsMutableSegmentsEvicted() {
+			return false, nil
+		}
+	case series.ColdWrite:
+		if !block.IsSealed() || !block.NeedsColdMutableSegmentsEvicted() {
```

Comment:
Sometimes the blocks will be sealed and sometimes they won't (i.e. cold writes into a mutable block). Need to think about this lifecycle.